iOS/Source/SplashViewController.swift

```diff
@@ -0,0 +1,157 @@
+import UIKit
+import CoreLocation
+import UserNotifications
+
+class SplashViewController: UIViewController {
+
+    lazy var locationTracker: LocationTracker = {
+        let tracker = LocationTracker()
+        tracker.delegate = self
+
+        // should this happen here? I don't think soooo
+//        Timer.scheduledTimer(timeInterval: 30, target: self, selector: #selector(self.updateDelegate), userInfo: nil, repeats: true)
+
+        return tracker
+    }()
+
+    private(set) var location: Location? {
+        set {
+            // make this a function in location make use of Codable please
+            if let location = newValue {
+                UserDefaults.standard.set(location.coordinates.latitude, forKey: "latitude")
+                UserDefaults.standard.set(location.coordinates.longitude, forKey: "longitude")
+                UserDefaults.standard.set(location.city, forKey: "city")
+                UserDefaults.standard.set(location.country, forKey: "country")
+            } else {
+                UserDefaults.standard.removeObject(forKey: "latitude")
+                UserDefaults.standard.removeObject(forKey: "city")
+                UserDefaults.standard.removeObject(forKey: "country")
+            }
+        }
+
+        get {
+            let latitude = UserDefaults.standard.object(forKey: "latitude") as? Double
+            let longitude = UserDefaults.standard.object(forKey: "longitude") as? Double
+            let city = UserDefaults.standard.object(forKey: "city") as? String
+            let country = UserDefaults.standard.object(forKey: "country") as? String
+
+            if let latitude = latitude, let longitude = longitude, let city = city, let country = country {
+                let coordinate = CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
+                return Location(coordinate: coordinate, city: city, country: country)
+            } else {
+                return nil
+            }
+        }
+    }
+
+    lazy var onboardingView: OnboardingView = {
+        let view = OnboardingView()
+        view.delegate = self
+
+        return view
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.view.backgroundColor = Theme.nightBackground
+        self.addObservers()
+
+        self.addSubViewsAndConstraints()
+        self.updateOnboardingStatus()
+    }
+
+    private func addSubViewsAndConstraints() {
+        self.view.addSubview(onboardingView)
+
+        onboardingView.edges(to: self.view)
+    }
+
+    func presentMainController(withLocation location: Location) {
+        let daylightModelController = DaylightModelController(location: location)
+        let mainController = MainController(withDaylightModelController: daylightModelController)
+        mainController.modalTransitionStyle = .crossDissolve
+        self.present(mainController, animated: true)
+    }
+
+    func addObservers() {
+        self.removeObservers()
```

Comment:
Is this necessary since this is only firing in `viewDidLoad`? It doesn't look like `removeObservers` is being called anywhere else in this class, either.